### ISSUE TITLE
E: Unable to locate package php-gettext during install

### ISSUE THREAD

**Issue by ChicagoJay** [FOGProject/fogproject#453]
Brand new VM, running Ubuntu 21.10.3, fully updated.

Install directory is `/opt/fog/`.

Trying to run `/opt/fog/fogproject-master/bin/installfog.sh` results: `E: Unable to locate package php-gettext`

Install and error logs attached.

[foginstall.zip](https://github.com/FOGProject/fogproject/files/7871889/foginstall.zip)

I think this has to do with trying to get a non-existent version of PHP.  The script seems to call for 7.1 but 8.1 is the current version.  I'm not sure how your script is installing PHP, but `apt install php` will always get the latest stable version.

**Comment by ChicagoJay** [comment]
I think I found it.  ~\lib\ubuntu\config.sh needs an entry for 21 under Ubuntu (line 46) that will get the correct version of PHP for Ubuntu 21.

**Comment by ChicagoJay** [comment]
So now the script tries to install PHP 7.4 (instead of 7.1).  Unfortunately, the script still dies because `php-gettext` is failing because php7.4 and modules aren't being found.  

Apparently, it's now called php-php-gettext.  (Really?!?)

In order to install 7.4 explicitly in Ubuntu 21, one needs to add the devs' repository: `add-apt-repository ppa:ondrej/php`

I'm not sure if FOG requires PHP 7, or if it will work with PHP 8, but if it will, then simply running apt install php and modules without the version should do the trick.

I modified the repository list at line 109 to include php-php-gettext instead of php-gettext.  I don't know if that will break other distros or not.

The code above has all the fixes.

**Comment by ChicagoJay** [comment]
I don't know if something is incompatible or not, but the install script now gets as far as the step where one needs to log into the web interface, to set up the schema.  However, when logging in, I get a 500 error from Apache.

*I punt*

**Comment by ChicagoJay** [comment]
I just read the issue about php-php-gettext.  Maybe I should try the devel branch....

**Comment by Sebastian-Roth** [comment]
@ChicagoJay Welcome to the FOG universe. You are on the right track now. FOG version 1.5.9 came out way before Ubuntu 21.10 was released and it's impossible to predict the packages that come with newer versions beforehand. We are trying to make the installer more universal but that will take a fair amount of work.

So back to your question. So yes, using `dev-branch` will help you install FOG on Ubuntu 21.10. BUT as it comes with PHP 8.0 there is another big hurdle. Check out the forum post here as well: https://forums.fogproject.org/topic/15914/install-fog-on-ubuntu-server-21-10-issues

We consider the short term Ubuntu releases as kind of "nice to have". The FOG dev team is small and there is lots of work to do to keep up with everything. You seem to be good with bash scripting and all that. Would you want to help taking care of that stuff?

**Comment by ChicagoJay** [comment]
> We consider the short term Ubuntu releases as kind of "nice to have". The FOG dev team is small and there is lots of work to do to keep up with everything. You seem to be good with bash scripting and all that. Would you want to help taking care of that stuff?

I'd love to, but I am quite busy with my day job!  ;)  Also, I don't really know HOW to help.  I've never posted code or pull requests before.  In fact, I only just learned what pull requests WERE a few months ago...

I actually watched Chuck develop the original FOG, back when he was at ATHS, in the early oughts.  I work maybe 20 minutes away from there, in a PK-8 school district.

Anyway - I spun up an Ubuntu 20.04LTS server, and the install completed!
